lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,328 @@
+                    BatchDeleteException, StopBeforeLambdaTimeoutException {
+        if (reportProcessingRequest.continueUserIdentityScan()) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Processing unique users to update their aggregate identities."));
+
+            List<ReportUserIdentityItem> totalReportUserIdentityItems = new ArrayList<>();
+            for (var page :
+                    reportUserIdentityScanDynamoDataStore.getItems(
+                            reportProcessingRequest.userIdentitylastEvaluatedKey(),
+                            ATTR_NAME_USER_ID,
+                            "identity",
+                            "migrated")) {
+                var userIds = page.items().stream().map(ReportUserIdentityItem::getUserId).toList();
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Checking user identity for total (%s) users.",
+                                        userIds.size())));
+                List<ReportUserIdentityItem> reportUserIdentityItems = new ArrayList<>();
+                for (String userId : userIds) {
+                    evaluateUserIdentityDetail(reportUserIdentityItems, userId);
+                }
+                LOGGER.info(LogHelper.buildLogMessage("Updating processed user's identity."));
+                reportUserIdentityScanDynamoDataStore.createOrUpdate(reportUserIdentityItems);
+
+                aggregateAndPersistIdentities(reportUserIdentityItems);
+
+                totalReportUserIdentityItems.addAll(reportUserIdentityItems);
+                reportProcessingResult.userIdentitylastEvaluatedKey(page.lastEvaluatedKey());
+
+                try {
+                    checkLambdaRemainingExecutionTime(context);
+                } catch (StopBeforeLambdaTimeoutException e) {
+                    reportProcessingResult.users(totalReportUserIdentityItems);
+                    throw e;
+                }
+            }
+            LOGGER.info(LogHelper.buildLogMessage("User identity check scan completed."));
+            reportProcessingResult.userIdentitylastEvaluatedKey(null);
+            reportProcessingResult.users(totalReportUserIdentityItems);
+        }
+    }
+
+    private void evaluateUserIdentityDetail(
+            List<ReportUserIdentityItem> reportUserIdentityItems, String userId)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException {
+        var tacticalVcs = verifiableCredentialService.getVcs(userId);
+        if (!userIdentityService.areVcsCorrelated(tacticalVcs)) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            String.format(
+                                    "User (%s) VCs not correlated.",
+                                    ReportUserIdentityItem.getUserHash(userId))));
```

Comment:
It'll be easier to process these logs if the message is fixed and the user hash is included as a separate field (e.g. `.with("user_hash", ...)`)

Reply:
done